relay-server/src/service.rs

```diff
@@ -120,20 +111,7 @@ impl ServiceState {
         };
 
         let buffer = Arc::new(BufferGuard::new(config.envelope_buffer_size()));
-        #[allow(unused_mut)]
-        let mut envelope_manager = EnvelopeManagerService::new(config.clone());
-
-        #[cfg(feature = "processing")]
-        if config.processing_enabled() {
-            let rt = create_runtime("store-rt", 1);
-            let store = StoreService::create(config.clone())?.start_in(&rt);
-            envelope_manager.set_store_forwarder(store);
-            _store_runtime = Some(rt);
-        }
-
-        let envelope_manager = envelope_manager.start();
-        let test_store = TestStoreService::new(config.clone()).start();
-
+        let (envelope_manager, envelope_manager_rx) = channel(EnvelopeProcessorService::name());
```

Comment:
I assume this is to resolve the circular dependency between manager and processor? Could you add a comment explaining this?

Reply:
Yeah, we have to get an address of the manager before we can use it in the processor, still we also have to have a processor address in the manager here.